app/models/template_input.rb

```diff
@@ -26,6 +26,7 @@ class UnsatisfiedRequiredInput < ::Foreman::Exception
   validates :variable_name, :presence => { :if => :variable_template_input? }
   validates :puppet_parameter_name, :puppet_class_name, :presence => { :if => :puppet_parameter_template_input? }
   validates :value_type, inclusion: { in: VALUE_TYPE }
+  validates :default, inclusion: { in: :possible_values, if: :options_defined? }
```

Comment:
I think we can simplify here and get rid of those two helper functions:
1. `if` is better out of the inclusion scope, we want to skip the whole validation
1. we can write inline proc for `if`, not to introduce method just for this if
1. we can use `:allow_blank => true`, what will skip the validation if the value is `.blank? => true`
```suggestion
  validates :default, inclusion: { in: :options_array }, if: -> { options.present? }, allow_blank: true
```

Reply:
Thanks, changed.

Reply:
Thanks, changed.